src/Nethermind/Nethermind.EofParser/LogInterceptor.cs

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Text;
+using Nethermind.Logging;
+public static class LogSink
+{
+    public static StringBuilder ContentBuilder { get; set; } = new StringBuilder();
+    public static string Content => ContentBuilder.ToString();
+    public static void Clear()
+    {
+        LogSink.ContentBuilder.Clear();
+    }
+}
+public class InMemoryLogger : ILogger
+{
+    public bool IsDebug { get; set; }
+    public bool IsInfo { get; set; }
+    public bool IsWarn { get; set; }
+    public bool IsError { get; set; }
+    public bool IsTrace { get; set; }
+
+
+    public void Trace(string message)
+    {
+        if (IsTrace)
+        {
+            LogSink.ContentBuilder.Append($"{message}\n\t");
+        }
+    }
+
+    void ILogger.Debug(string text) => throw new NotImplementedException();
+
+    void ILogger.Error(string text, Exception ex) => throw new NotImplementedException();
+
+    void ILogger.Info(string text) => throw new NotImplementedException();
+
+    void ILogger.Warn(string text) => throw new NotImplementedException();
```

Comment:
would not leave those as exceptions

Reply:
Even if they won't be used?

Reply:
They are not used now, but they can be used later - if you leave it at that you break `Liskov substitution principle`.